### PR TITLE
Multiple authentication context classes for SAML IDP

### LINKED ISSUE #12887: Multiple Authentication Context Classes for SAML IDP
**Is your feature request related to a problem? Please describe.**

For the federated SAML IDP only one AuthnContextClassRef can be added in the current implementation. But according to SAML2.0 specification it supports multiple AuthnContextClassRefs to include in SAML AuthnRequest. 
As an example there are scenarios where third party IDPs which configured as federated IDPs in WSO2 IS allows authenticating using multiple AuthnContextClassRefs and the third party IDP expects Exact as the authentication comparison level. In such scenarios the request may fail from the third party IDP side.

**Describe the solution you would prefer**

Add capability to add multiple authentication context classes to SAML IDP from UI and include multiple AuthnContextClassRefs in SAML AuthnRequest

### PR DESCRIPTION
**Pull request by Dinithi-Hasanika**
Fixes https://github.com/wso2/product-is/issues/12887

Improved to use Multiple authentication context classes for SAML IDP so that multiple authentication context classes can be included in SAMLAuthnRequest

### CODE REVIEW

File: components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/manager/DefaultSAML2SSOManager.java [hunk 1]
```diff
@@ -896,28 +896,37 @@ protected RequestedAuthnContext buildRequestedAuthnContext(AuthnRequest inboundA
...
-                String samlAuthnContextURN = IdentityApplicationManagementUtil
-                        .getSAMLAuthnContextClasses().get(authnContextClass);
-                if (!StringUtils.isBlank(samlAuthnContextURN)) {
-                    //There was one matched URN for give authnContextClass.
-                    authnContextClassRef.setAuthnContextClassRef(samlAuthnContextURN);
-                } else {
-                    //There are no any matched URN for given authnContextClass, so added authnContextClass name to the
-                    // AuthnContextClassRef.
-                    authnContextClassRef.setAuthnContextClassRef(authnContextClass);
+                String[] authnContextClassList = authnContextClass.split(",");
```
Comment (maneeshaindrachapa) [inline components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/manager/DefaultSAML2SSOManager.java]: Can we add a constant for `,` ?

File: components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/manager/DefaultSAML2SSOManager.java [hunk 2]
```diff
@@ -896,28 +896,37 @@ protected RequestedAuthnContext buildRequestedAuthnContext(AuthnRequest inboundA
...
+                for (String authnContextClassListElement : authnContextClassList) {
+                    AuthnContextClassRef authnContextClassRef = authnContextClassRefBuilder
+                            .buildObject(SAMLConstants.SAML20_NS,
+                                    AuthnContextClassRef.DEFAULT_ELEMENT_LOCAL_NAME,
+                                    SAMLConstants.SAML20_PREFIX);
+                    String samlAuthnContextURN = IdentityApplicationManagementUtil
+                            .getSAMLAuthnContextClasses().get(authnContextClassListElement);
+                    if (!StringUtils.isBlank(samlAuthnContextURN)) {
+                        //There was one matched URN for given authnContextClass.
```
Comment (maneeshaindrachapa) [inline components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/manager/DefaultSAML2SSOManager.java]: Add space before adding a comment. ex: `// There was one matched URN for given authnContextClass.`

File: components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/manager/DefaultSAML2SSOManager.java [hunk 3]
```diff
@@ -896,28 +896,37 @@ protected RequestedAuthnContext buildRequestedAuthnContext(AuthnRequest inboundA
...
-                    authnContextClassRef.setAuthnContextClassRef(authnContextClass);
+                String[] authnContextClassList = authnContextClass.split(",");
+                for (String authnContextClassListElement : authnContextClassList) {
+                    AuthnContextClassRef authnContextClassRef = authnContextClassRefBuilder
+                            .buildObject(SAMLConstants.SAML20_NS,
+                                    AuthnContextClassRef.DEFAULT_ELEMENT_LOCAL_NAME,
+                                    SAMLConstants.SAML20_PREFIX);
+                    String samlAuthnContextURN = IdentityApplicationManagementUtil
+                            .getSAMLAuthnContextClasses().get(authnContextClassListElement);
+                    if (!StringUtils.isBlank(samlAuthnContextURN)) {
```
Comment (maneeshaindrachapa) [inline components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/manager/DefaultSAML2SSOManager.java]: Can we use `StringUtils.isNotBlank()` method?

File: components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/manager/DefaultSAML2SSOManager.java (hunk 1 above)
Comment (Dinithi-Hasanika) [reply components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/manager/DefaultSAML2SSOManager.java]: added a constant for this

File: components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/manager/DefaultSAML2SSOManager.java (hunk 2 above)
Comment (Dinithi-Hasanika) [reply components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/manager/DefaultSAML2SSOManager.java]: added

File: components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/manager/DefaultSAML2SSOManager.java (hunk 3 above)
Comment (Dinithi-Hasanika) [reply components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/manager/DefaultSAML2SSOManager.java]: yes. resolved with this method

**Review by pamodaaw — APPROVED** [review]

**Pull request merged**